Hi, i added same patch as referenced in: https://github.com/huggingface/transformers/pull/45916/changes

The issue, it keeps crashing by keys, seems like pandas too changed indexes to labels, i made this changes too but i don't know the real impact it could have. Actually it works but could affect something else.

I gonna point too the same part that was mentioned about `ArrowStringArray`. Adding the provided fix, it seems in `C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\pandas\core\internals\managers.py` now it uses `NumpyExtensionArray` which is different and having `dtype('O')`, referencing to line `445`, this flow is performed successful till the point when uses `column indexes` instead of `labels in keys`.

Causing next stacktrace:

```
(.venv) C:\Users\usuario\Documents\aibased>C:/Users/usuario/AppData/Local/Programs/Python/Python313/python.exe c:/Users/usuario/Documents/aibased/main.py
Warning: You are sending unauthenticated requests to the HF Hub. Please set a HF_TOKEN to enable higher rate limits and faster downloads.
Loading weights: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 403/403 [00:00<00:00, 14479.97it/s]
Traceback (most recent call last):
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\pandas\core\indexes\base.py", line 3641, in get_loc
    return self._engine.get_loc(casted_key)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^
  File "pandas/_libs/index.pyx", line 168, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 176, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 583, in pandas._libs.index.StringObjectEngine._check_type
KeyError: 0

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "c:\Users\usuario\Documents\aibased\main.py", line 15, in <module>
    print(tqa(table=table, query=question)['cells'][0])
          ~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\pipelines\table_question_answering.py", line 285, in __call__
    results = super().__call__(pipeline_inputs, **kwargs)
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\pipelines\base.py", line 1247, in __call__
    outputs = list(final_iterator)
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\pipelines\pt_utils.py", line 126, in __next__
    item = next(self.iterator)
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\pipelines\pt_utils.py", line 126, in __next__
    item = next(self.iterator)
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\torch\utils\data\dataloader.py", line 741, in __next__
    data = self._next_data()
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\torch\utils\data\dataloader.py", line 801, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\torch\utils\data\_utils\fetch.py", line 54, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
            ~~~~~~~~~~~~^^^^^
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\pipelines\pt_utils.py", line 19, in __getitem__
    processed = self.process(item, **self.params)
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\pipelines\table_question_answering.py", line 321, in preprocess
    inputs = self.tokenizer(table, query, return_tensors="pt", truncation=truncation, padding=padding)
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\models\tapas\tokenization_tapas.py", line 619, in __call__
    return self.encode_plus(
           ~~~~~~~~~~~~~~~~^
        table=table,
        ^^^^^^^^^^^^
    ...<17 lines>...
        **kwargs,
        ^^^^^^^^^
    )
    ^
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\models\tapas\tokenization_tapas.py", line 982, in encode_plus
    return self._encode_plus(
           ~~~~~~~~~~~~~~~~~^
        table=table,
        ^^^^^^^^^^^^
    ...<16 lines>...
        **kwargs,
        ^^^^^^^^^
    )
    ^
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\models\tapas\tokenization_tapas.py", line 1034, in _encode_plus
    return self.prepare_for_model(
           ~~~~~~~~~~~~~~~~~~~~~~^
        table,
        ^^^^^^
    ...<17 lines>...
        verbose=verbose,
        ^^^^^^^^^^^^^^^^
    )
    ^
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\models\tapas\tokenization_tapas.py", line 1170, in prepare_for_model
    raw_table = add_numeric_table_values(raw_table)
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\models\tapas\tokenization_tapas.py", line 2778, in add_numeric_table_values
    _get_column_values(table, col_index),
    ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\models\tapas\tokenization_tapas.py", line 2694, in _get_column_values
    text = normalize_for_match(row[col_index].text) #Patch 12052026
                               ~~~^^^^^^^^^^^
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\pandas\core\series.py", line 959, in __getitem__
    return self._get_value(key)
           ~~~~~~~~~~~~~~~^^^^^
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\pandas\core\series.py", line 1046, in _get_value
    loc = self.index.get_loc(label)
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\pandas\core\indexes\base.py", line 3648, in get_loc
    raise KeyError(key) from err
KeyError: 0

```

After that was applied new change which is not definitive and must be validate by tests but avoids this first issue.

Location: `C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\models\tapas\tokenization_tapas.py`

Line: 2694

```
def _get_column_values(table, col_index):
    """
    Parses text in column and returns a dict mapping row_index to values. This is the _get_column_values function from
    number_annotation_utils.py of the original implementation

    Args:
      table: Pandas dataframe
      col_index: integer, indicating the index of the column to get the numeric values of
    """
    index_to_values = {}
    for row_index, row in table.iterrows():
        #text = normalize_for_match(row[col_index].text) #Patch 12052026 Commenting affected part.
        text = normalize_for_match(row[row.keys()[col_index]].text) #New part not definitive.
        index_to_values[row_index] = list(_get_numeric_values(text))
    return index_to_values

```

After fix this first part, here is a new raised exception related to index usage instead of labels as was mentioned, stacktrace is this one:

```
(.venv) C:\Users\usuario\Documents\aibased>C:/Users/usuario/AppData/Local/Programs/Python/Python313/python.exe c:/Users/usuario/Documents/aibased/main.py
Warning: You are sending unauthenticated requests to the HF Hub. Please set a HF_TOKEN to enable higher rate limits and faster downloads.
Loading weights: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 403/403 [00:00<00:00, 14705.08it/s]
Traceback (most recent call last):
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\pandas\core\indexes\base.py", line 3641, in get_loc
    return self._engine.get_loc(casted_key)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^
  File "pandas/_libs/index.pyx", line 168, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 176, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 583, in pandas._libs.index.StringObjectEngine._check_type
KeyError: 0

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "c:\Users\usuario\Documents\aibased\main.py", line 15, in <module>
    print(tqa(table=table, query=question)['cells'][0])
          ~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\pipelines\table_question_answering.py", line 285, in __call__
    results = super().__call__(pipeline_inputs, **kwargs)
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\pipelines\base.py", line 1247, in __call__
    outputs = list(final_iterator)
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\pipelines\pt_utils.py", line 126, in __next__
    item = next(self.iterator)
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\pipelines\pt_utils.py", line 126, in __next__
    item = next(self.iterator)
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\torch\utils\data\dataloader.py", line 741, in __next__
    data = self._next_data()
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\torch\utils\data\dataloader.py", line 801, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\torch\utils\data\_utils\fetch.py", line 54, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
            ~~~~~~~~~~~~^^^^^
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\pipelines\pt_utils.py", line 19, in __getitem__
    processed = self.process(item, **self.params)
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\pipelines\table_question_answering.py", line 321, in preprocess
    inputs = self.tokenizer(table, query, return_tensors="pt", truncation=truncation, padding=padding)
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\models\tapas\tokenization_tapas.py", line 619, in __call__
    return self.encode_plus(
           ~~~~~~~~~~~~~~~~^
        table=table,
        ^^^^^^^^^^^^
    ...<17 lines>...
        **kwargs,
        ^^^^^^^^^
    )
    ^
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\models\tapas\tokenization_tapas.py", line 982, in encode_plus
    return self._encode_plus(
           ~~~~~~~~~~~~~~~~~^
        table=table,
        ^^^^^^^^^^^^
    ...<16 lines>...
        **kwargs,
        ^^^^^^^^^
    )
    ^
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\models\tapas\tokenization_tapas.py", line 1034, in _encode_plus
    return self.prepare_for_model(
           ~~~~~~~~~~~~~~~~~~~~~~^
        table,
        ^^^^^^
    ...<17 lines>...
        verbose=verbose,
        ^^^^^^^^^^^^^^^^
    )
    ^
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\models\tapas\tokenization_tapas.py", line 1175, in prepare_for_model
    column_ranks, inv_column_ranks = self._get_numeric_column_ranks(column_ids, row_ids, raw_table)
                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\models\tapas\tokenization_tapas.py", line 1510, in _get_numeric_column_ranks
    table_numeric_values = self._get_column_values(table, col_index)
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\models\tapas\tokenization_tapas.py", line 1490, in _get_column_values
    cell = row[col_index] #Patch 12052026
           ~~~^^^^^^^^^^^
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\pandas\core\series.py", line 959, in __getitem__
    return self._get_value(key)
           ~~~~~~~~~~~~~~~^^^^^
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\pandas\core\series.py", line 1046, in _get_value
    loc = self.index.get_loc(label)
  File "C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\pandas\core\indexes\base.py", line 3648, in get_loc
    raise KeyError(key) from err
KeyError: 0
```

Performed change for avoid the whole issue which could be not definitive for some non desired impact:

Location: `C:\Users\usuario\AppData\Local\Programs\Python\Python313\Lib\site-packages\transformers\models\tapas\tokenization_tapas.py`

Line: 1490

```
def _get_column_values(self, table, col_index):
        table_numeric_values = {}
        for row_index, row in table.iterrows():
            #cell = row[col_index] #Patch 12052026  Commenting affected part.
            cell = row[row.keys()[col_index]]
            if cell.numeric_value is not None:
                table_numeric_values[row_index] = cell.numeric_value
        return table_numeric_values
```

After this execution works as expected for my specific use case.

```
(.venv) C:\Users\usuario\Documents\aibased>C:/Users/usuario/AppData/Local/Programs/Python/Python313/python.exe c:/Users/usuario/Documents/aibased/main.py
Loading weights: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 403/403 [00:00<00:00, 12101.18it/s]
53
```

--- a/src/transformers/models/tapas/tokenization_tapas.py
+++ b/src/transformers/models/tapas/tokenization_tapas.py
@@ -1487,7 +1487,7 @@ class TapasTokenizer(PreTrainedTokenizer):
     def _get_column_values(self, table, col_index):
         table_numeric_values = {}
         for row_index, row in table.iterrows():
-            cell = row[col_index]
+            cell = row.iloc[col_index]
             if cell.numeric_value is not None:
                 table_numeric_values[row_index] = cell.numeric_value
         return table_numeric_values
@@ -2690,7 +2690,7 @@ def _get_column_values(table, col_index):
     """
     index_to_values = {}
     for row_index, row in table.iterrows():
-        text = normalize_for_match(row[col_index].text)
+        text = normalize_for_match(row.iloc[col_index].text)
         index_to_values[row_index] = list(_get_numeric_values(text))
     return index_to_values
 

--- a/tests/models/tapas/test_tokenization_tapas.py
+++ b/tests/models/tapas/test_tokenization_tapas.py
@@ -274,25 +274,24 @@ class TapasTokenizationTest(TokenizerTesterMixin, unittest.TestCase):
             [tokenizer.tokenize(t) for t in ["Test", "\xad", "test"]], [["[UNK]"], ["[EMPTY]"], ["[UNK]"]]
         )
 
-    def test_add_numeric_table_values_with_pyarrow_string_dtype(self):
-        # Regression test for https://github.com/huggingface/transformers/issues/45901: in pandas
-        # 3.x string columns default to a pyarrow-backed StringDtype that rejects writes of
-        # arbitrary Python objects (it calls len() on the value), breaking the Cell insertion
-        # inside add_numeric_table_values. Simulate that default with future.infer_string.
-        from transformers.models.tapas.tokenization_tapas import Cell, add_numeric_table_values
-
+    def test_tokenize_table_with_pyarrow_string_dtype(self):
+        # Regression test for https://github.com/huggingface/transformers/issues/45901. On pandas
+        # 3.x, string columns default to a pyarrow-backed StringDtype, which (a) rejects writes
+        # of Cell objects inside add_numeric_table_values, and (b) makes positional row[col_index]
+        # access raise KeyError in the downstream _get_column_values helpers. Simulate the 3.x
+        # default with future.infer_string and drive the full tokenizer end-to-end.
+        tokenizer = self.get_tokenizer()
         data = {"Actors": ["Brad Pitt", "Leonardo Di Caprio"], "Number of movies": ["87", "53"]}
         original = pd.get_option("future.infer_string")
         pd.set_option("future.infer_string", True)
         try:
             table = pd.DataFrame.from_dict(data)
-            result = add_numeric_table_values(table)
+            encoding = tokenizer(table, "how many movies does Leonardo Di Caprio have?")
         finally:
             pd.set_option("future.infer_string", original)
 
-        self.assertIsInstance(result.iloc[0, 0], Cell)
-        self.assertEqual(result.iloc[1, 1].text, "53")
-        self.assertEqual(result.iloc[1, 1].numeric_value.float_value, 53.0)
+        self.assertIn("input_ids", encoding)
+        self.assertGreater(len(encoding["input_ids"]), 0)
 
     @slow
     def test_sequence_builders(self):

